src/lib/transaction_snark/test/zkapps_examples/thing/thing.ml

```diff
@@ -0,0 +1,183 @@
+  }
+
+(* why are there two updates in there? an [account_update] and then a [deploy_account_update] *)
+
+let account_updates =
+  []
+  (* this looks like a call to prover *)
+  |> Zkapp_command.Call_forest.cons_tree account_update
+  (* and this like a normal account update call *)
+  |> Zkapp_command.Call_forest.cons deploy_account_update
+
+let memo = Signed_command_memo.empty
+
+(* what's a transaction commitment? (looks like it's just a field element which is a hash (poseidon) of the call forest (which is just `hash([prev_hash, new_el])`)) *)
+
+let transaction_commitment : Zkapp_command.Transaction_commitment.t =
+  (* TODO: This is a pain. *)
+  let account_updates_hash = Zkapp_command.Call_forest.hash account_updates in
+  Zkapp_command.Transaction_commitment.create ~account_updates_hash
+
+(* the one paying the fee is [pk_compressed] *)
+
+let fee_payer =
+  (* TODO: This is a pain. *)
+  { Account_update.Fee_payer.body =
+      { Account_update.Body.Fee_payer.dummy with
+        public_key = pk_compressed
+      ; fee = Currency.Fee.(of_nanomina_int_exn 100)
+      }
+  ; authorization = Signature.dummy
+  }
+
+let full_commitment =
+  (* TODO: This is a pain. *)
+  Zkapp_command.Transaction_commitment.create_complete transaction_commitment
+    ~memo_hash:(Signed_command_memo.hash memo)
+    ~fee_payer_hash:
+      (Zkapp_command.Digest.Account_update.create
+         (Account_update.of_fee_payer fee_payer) )
+
+(* TODO: Make this better. *)
+let sign_all ({ fee_payer; account_updates; memo } : Zkapp_command.t) :
+    Zkapp_command.t =
+  let fee_payer =
+    match fee_payer with
+    | { body = { public_key; _ }; _ }
+      when Public_key.Compressed.equal public_key pk_compressed ->
+        { fee_payer with
+          authorization =
+            Schnorr.Chunked.sign sk
+              (Random_oracle.Input.Chunked.field full_commitment)
+        }
+    | fee_payer ->
+        fee_payer
+  in
+  let account_updates =
+    Zkapp_command.Call_forest.map account_updates ~f:(function
+      | ({ body = { public_key; use_full_commitment; _ }
+         ; authorization = Signature _
+         } as account_update :
+          Account_update.t )
+        when Public_key.Compressed.equal public_key pk_compressed ->
+          let commitment =
+            if use_full_commitment then full_commitment
+            else transaction_commitment
+          in
+          { account_update with
+            authorization =
+              Control.Signature
+                (Schnorr.Chunked.sign sk
+                   (Random_oracle.Input.Chunked.field commitment) )
+          }
+      | account_update ->
+          account_update )
+  in
+  { fee_payer; account_updates; memo }
+
+let zkapp_command : Zkapp_command.t =
+  sign_all
+    { fee_payer = { body = fee_payer.body; authorization = Signature.dummy }
+    ; account_updates
+    ; memo
+    }
+
+let () =
+  (* create a fake ledger and do something with it ? *)
```

Comment:
Create a temporary ledger, pass it to the given function, then tidy it up once the function finishes.

If you think this documentation is useful, it should live on that function, not here.